### ISSUE TITLE
Tracing out systems with dimensions 1 on multipilcation creates dimention missmatch errors

### ISSUE THREAD

**Issue by tacruc** [qutip/qutip#823]
I'm woundering what is the reason for doing this partial trace
https://github.com/qutip/qutip/blob/0b068fc4876727303644c42c181af82e1f089700/qutip/qobj.py#L490
becauseit leads to errors doing:
```
O=tensor(sigmax(), qeye(1))
Psi = tensor(basis(2,0),basis(1,0))
O*O*Psi
```

would it be possible to implement an Option in settings to dissable this behavior? I would be willing to provide a PR.

**Comment by IIAOPSW** [comment]
AFAIK this error only manifests if you have a space of dimension 1.
and a space of dimension 1 is trivial. 

Is there a use case where this actually manifests?

**Comment by tacruc** [comment]
I use the qutip to do some Grover simulations. having a Hilbertspace of Grover an an additional entangled Hilbert space. In a simpler version there is only one state in the entangled Space so the program automatically chooses dimension 1. 
And then there are allot of errors coming because of this qutip error. 
At the moment I fixed it using a patched qutip version.

**Comment by ajgpitch** [comment]
Arne,

Please propose your patch in a PR on our Github. So long as it is
mathematically sound, then the context does not matter in this case.

If you need any advice on how to do this, then let us know. Email me direct
if you prefer. alex.pitchford@gmail.com

Thanks

Alex

**Comment by tacruc** [comment]
This issue can be closed since the PR #886 is merged or?
